graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -78,24 +74,12 @@ def _validate_macro_ast_directives(ast, inside_fold_scope=False):
         inside_fold_scope: bool, whether the subtree is within a @fold scope
     """
     subselection_inside_fold_scope = inside_fold_scope
-    directives_with_no_restrictions = frozenset({
-        FilterDirective.name,
-        TagDirective.name,
-        OptionalDirective.name,
-        RecurseDirective.name,
-        MacroEdgeDefinitionDirective.name,
-    })
-    disallowed_directives = frozenset({
-        OutputDirective.name,
-        OutputSourceDirective.name,
-    })
     for directive in ast.directives:
         name = directive.name.value
-        if name in directives_with_no_restrictions:
+        if name in DIRECTIVES_ALLOWED_IN_MACRO_EDGE_DEFINITION:
+            pass
+        elif name == MacroEdgeDefinitionDirective.name:
             pass
-        elif name in disallowed_directives:
-            raise GraphQLInvalidMacroError(u'Macros are not allowed to use the {} directive. '
-                                           u'Found usage {}'.format(name, directive))
         elif name == FoldDirective.name:
```

Comment:
Same as above,

Reply:
Now that the directive list has been renamed to OPTIONAL_DIRECTIVES_UNRESTRICTED_..., FoldDirective clearly doesn't belong in the set, as right here we check whether certain usage conditions for FoldDirective are met.